### PR TITLE
Unify k8s ip pool (restore cluster-cidr to previous value)

### LINKED ISSUE #741: Kubernetes simple-policy tutorial inconsistent results with service on release candidate
tl;dr I saw inconsistent results when accessing a service IP.

This occurred when going through the tutorial on the 2.2 release candidate at v2.2/getting-started/kubernetes/tutorials/simple-policy.
I saw this behavior after:
- enabling isolation
- adding the policy to allow access

Trying to access the nginx service with `wget -q --timeout=5 nginx -O -` resulted in a successful command 3 out of 10 attempts.

I looked up the service IP address and used that to eliminate the DNS service and the results were inconsistent still.

I tried wget using the IP addresses of the 2 pods backing the service and was able to consistently access them.  So it looks like the issue is with the service IP itself.

### PR DESCRIPTION
**Pull request by tmjd**
Fixes #741 
The `cluster-cidr` expected by the K8s etcd manifests is `192.168.0.0/16` (the default if no `cluster-cidr`) is specified.  A value of `10.244.0.0/16` was explicitly set which broke accessing a k8s service when using the etcd manifests.  This change restores that value to the default (though sets it explicitly) and changes the kdd manifests to match that value.

### CODE REVIEW

**Comment by tmjd** [comment]
These commits need squashing but I left them separate in case we didn't want to make this change to v2.1 and v2.0 but I don't expect that to be the case.

**Comment by djosborne** [comment]
Can you update the description of this to explain how this fixes #741 ? It's not clear to me why this is  needed.

**Comment by tmjd** [comment]
@djosborne PTAL

**Comment by djosborne** [comment]
@tmjd So from what I understand, there's places / values for this field:
1. The default  `--cluster-cidr` flag
2. The user-overridden `--cluster-cidr` flag 
3. The value of the calico pool.

A little history - I think 2 was originally set to `192.168.0.0/16` because ever since calico's first container implementation, that was the default pool in Calico. So we figured "modify kubernetes to use our default pool" was the way to go. But in actuality, modifiying Calico to use Kubernetes default is more practical.

So I'm concerned that we're changing 3 to be the value we chose for 2 for our demos, but in reality both should match whatever 1 is.

What is the value for 1? Does it depend on the deployment method? What does kubeadm set it by default? Is it ever random?

**Comment by tmjd** [comment]
@djosborne and I chatted so here is some info to leave a paper trail

The default in kubernetes when no `--cluster-cidr` is specified is `192.168.0.0/16`.
If using kubeadm the default when no `--pod-network-cidr` is specified is to start the kubernetes components without specifying a `--cluster-cidr` and therefore results in `192.168.0.0/16`.

So I believe with the change here we are matching our default IP Pool with the default `--cluster-cidr` for kubernetes and kubeadm.  We are now explicitly setting `--cluster-cidr` which should protect us from breakage in the future if the default changes.

**Comment by tmjd** [comment]
@djosborne PTAL

**Comment by tmjd** [comment]
Ok So I double checked and my memory certainly seems incorrect.  I removed `--cluster-cidr` from all the k8s components and did not have functioning services with that setup.  So It does look like the `cluster-cidr` is just unset when if it is not specified.

**Comment by tmjd** [comment]
I still think this is the correct change at this time since we had been using `192.168.0.0/16` for our k8s etcd manifests.

**Comment by djosborne** [comment]
@caseydavenport knows the right path here. I think the only reason I involved myself was he was OOO. Now that he's back, I think it's best to let him review this and clear the confusion.

**Comment by caseydavenport** [comment]
A quick summary of where I've gotten to:

I don't think we should be applying this change to v2.0 or v2.1 KDD manifests.  Those are set to `10.244.0.0/16` for a good reason - compatibility with the default flannel self-hosted manifest (since they provide policy only and need flannel or something else for networking).

In v2.2 (and master) the KDD manifest now includes routing, which means that by default it isn't meant to use with the kube-flannel manifest*, and so it should be OK to adjust the cluster-cidr to match what our etcd based manifests (also policy+routing) have used for a long time.

To be clear, the reason for unifying the value used for etcd+kdd is simply so that they both work on our vagrant guide, which sets `--cluster-cidr=192.168.0.0/16`. 

This change is related to, and will need to be considered when addressing https://github.com/projectcalico/calico/issues/750.

\* For interested readers, see [the Canal manifest](https://github.com/projectcalico/canal/blob/master/k8s-install/README.md) for the recommended all-in-one combination of Calico + flannel

File: v2.0/getting-started/kubernetes/installation/hosted/k8s-backend/index.md [hunk 1]
```diff
@@ -18,7 +18,7 @@ You must have a cluster which meets the following requirements:
 
 - You have a Kubernetes cluster configured to use CNI network plugins (i.e. by passing `--network-plugin=cni` to the kubelet)
 - Your Kubernetes controller manager is configured to allocate pod CIDRs (i.e. by passing `--allocate-node-cidrs=true` to the controller manager)
-- Your Kubernetes controller manager has been provided a cluster-cidr (i.e. by passing `--cluster-cidr=10.244.0.0/16`, which the manifest expects by default).
+- Your Kubernetes controller manager has been provided a cluster-cidr (i.e. by passing `--cluster-cidr=192.168.0.0/16`, which the manifest expects by default).
```
Comment (caseydavenport) [inline v2.0/getting-started/kubernetes/installation/hosted/k8s-backend/index.md]: I think we need to undo this change (and similar ones in v2.0 and v2.1)

**Comment by caseydavenport** [comment]
Once the v2.1 and v2.0 stuff is backed out then I think this is good to merge.

**Comment by caseydavenport** [comment]
LGTM
